ccdm: flow-client build failed on windows machine

Node version: 12.8.0
OS: Windows 10 (works with Mac OS)

run `mvn clean install -DskipTests`

the build failed on `flow-client` module, the error log is listed:

```
[INFO] --- gwt-maven-plugin:2.8.2:compile (default) @ flow-client ---
[INFO] Compiling module com.vaadin.ClientEngine
[INFO]    Warnings in com/vaadin/client/flow/dom/DomElement.java
[INFO]       [WARN] Line 119: [unusable-by-js] Return type of 'Element DomElement.getFirstElementChild()' is not usable by but exposed to JavaScript.
[INFO]       [WARN] Line 127: [unusable-by-js] Return type of 'Element DomElement.getLastElementChild()' is not usable by but exposed to JavaScript.
[INFO]       [WARN] Line 153: [unusable-by-js] Return type of 'HTMLCollection DomElement.getChildren()' is not usable by but exposed to JavaScript.
[INFO]       [WARN] Line 163: [unusable-by-js] Return type of 'Element DomElement.querySelector(String)' is not usable by but exposed to JavaScript.
[INFO]    Warnings in com/vaadin/client/flow/dom/DomNode.java
[INFO]       [WARN] Line 61: [unusable-by-js] Return type of 'Node DomNode.getFirstChild()' is not usable by but exposed to JavaScript.
[INFO]       [WARN] Line 69: [unusable-by-js] Return type of 'Node DomNode.getLastChild()' is not usable by but exposed to JavaScript.
[INFO]       [WARN] Line 77: [unusable-by-js] Return type of 'Node DomNode.getNextSibling()' is not usable by but exposed to JavaScript.
[INFO]       [WARN] Line 85: [unusable-by-js] Return type of 'Node DomNode.getPreviousSibling()' is not usable by but exposed to JavaScript.
[INFO]       [WARN] Line 110: [unusable-by-js] Type of parameter 'node' in 'void DomNode.appendChild(Node)' is not usable by but exposed to JavaScript.
[INFO]       [WARN] Line 121: [unusable-by-js] Type of parameter 'newChild' in 'void DomNode.insertBefore(Node, Node)' is not usable by but exposed to JavaScript.
[INFO]       [WARN] Line 121: [unusable-by-js] Type of parameter 'refChild' in 'void DomNode.insertBefore(Node, Node)' is not usable by but exposed to JavaScript.
[INFO]       [WARN] Line 130: [unusable-by-js] Type of parameter 'childNode' in 'void DomNode.removeChild(Node)' is not usable by but exposed to JavaScript.
[INFO]       [WARN] Line 142: [unusable-by-js] Type of parameter 'newChild' in 'void DomNode.replaceChild(Node, Node)' is not usable by but exposed to JavaScript.
[INFO]       [WARN] Line 142: [unusable-by-js] Type of parameter 'oldChild' in 'void DomNode.replaceChild(Node, Node)' is not usable by but exposed to JavaScript.
[INFO]       [WARN] Line 154: [unusable-by-js] Return type of 'Node DomNode.cloneNode(boolean)' is not usable by but exposed to JavaScript.
[INFO]       [WARN] Line 163: [unusable-by-js] Return type of 'Node DomNode.getParentNode()' is not usable by but exposed to JavaScript.
[INFO]    Warnings in com/vaadin/client/flow/dom/PolymerDomApiImpl.java
[INFO]       [WARN] Line 68: [unusable-by-js] Type of parameter 'node' in 'DomElement PolymerDomApiImpl.Polymer.dom(Node)' is not usable by but exposed to JavaScript.
[INFO]    [WARN] Suppress "[unusable-by-js]" warnings by adding a `@SuppressWarnings("unusable-by-js")` annotation to the corresponding member.
[INFO]    Compiling 1 permutation
[INFO]       Compiling permutation 0...
[INFO]    Compile of permutations succeeded
[INFO]    Compilation succeeded -- 18.299s
[INFO] Linking into C:\WorkFolder\flow\flow-client\target\classes\META-INF\resources\VAADIN\static\client; Writing extras to C:\WorkFolder\flow\flow-client\target\extra\client
[INFO]    Link succeeded
[INFO]    Linking succeeded -- 0.904s
[INFO]
[INFO] --- gwt-maven-plugin:2.8.2:generateAsync (default) @ flow-client ---
[INFO]
[INFO] --- exec-maven-plugin:1.6.0:exec (typescript-compile) @ flow-client ---

> @vaadin/flow-deps@0.0.1 compile C:\WorkFolder\flow\flow-client
> npm run build && npm run lint


> @vaadin/flow-deps@0.0.1 build C:\WorkFolder\flow\flow-client
> npm run client && tsc


> @vaadin/flow-deps@0.0.1 client C:\WorkFolder\flow\flow-client
> npm run _cp_to_src && npm run _wrap_fnc && npm run _cp_to_tgt


> @vaadin/flow-deps@0.0.1 _cp_to_src C:\WorkFolder\flow\flow-client
> ncp target/classes/META-INF/resources/VAADIN/static/client/client-*.cache.js src/main/resources/META-INF/resources/frontend/FlowClient.js

There were errors during the copy.
Error: ENOENT: no such file or directory, lstat 'C:\WorkFolder\flow\flow-client\target\classes\META-INF\resources\VAADIN\static\client\client-*.cache.js'
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @vaadin/flow-deps@0.0.1 _cp_to_src: `ncp target/classes/META-INF/resources/VAADIN/static/client/client-*.cache.js src/main/resources/META-INF/resources/frontend/FlowClient.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @vaadin/flow-deps@0.0.1 _cp_to_src script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\zhe\AppData\Roaming\npm-cache\_logs\2020-01-07T09_33_42_868Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @vaadin/flow-deps@0.0.1 client: `npm run _cp_to_src && npm run _wrap_fnc && npm run _cp_to_tgt`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @vaadin/flow-deps@0.0.1 client script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\zhe\AppData\Roaming\npm-cache\_logs\2020-01-07T09_33_42_889Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @vaadin/flow-deps@0.0.1 build: `npm run client && tsc`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @vaadin/flow-deps@0.0.1 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\zhe\AppData\Roaming\npm-cache\_logs\2020-01-07T09_33_42_914Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @vaadin/flow-deps@0.0.1 compile: `npm run build && npm run lint`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @vaadin/flow-deps@0.0.1 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\zhe\AppData\Roaming\npm-cache\_logs\2020-01-07T09_33_42_936Z-debug.log
[ERROR] Command execution failed.
org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
        at org.apache.commons.exec.DefaultExecutor.executeInternal(DefaultExecutor.java:404)
        at org.apache.commons.exec.DefaultExecutor.execute(DefaultExecutor.java:166)
        at org.codehaus.mojo.exec.ExecMojo.executeCommandLine(ExecMojo.java:804)
        at org.codehaus.mojo.exec.ExecMojo.executeCommandLine(ExecMojo.java:751)
        at org.codehaus.mojo.exec.ExecMojo.execute(ExecMojo.java:313)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:309)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:194)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:107)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:993)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:345)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:191)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

--- a/flow-client/package.json
+++ b/flow-client/package.json
@@ -10,10 +10,7 @@
   },
   "scripts": {
     "lint": "eslint 'src/main/resources/META-INF/resources/frontend/*.js' && tslint 'src/main/resources/META-INF/resources/frontend/*.ts'",
-    "_cp_to_src": "ncp target/classes/META-INF/resources/VAADIN/static/client/client-*.cache.js src/main/resources/META-INF/resources/frontend/FlowClient.js",
-    "_wrap_fnc": "replace-in-files --regex '(.+[\\s\\S]*)' --replacement 'const init = function(){\\n$1\\n};\\nexport {init};\\n' src/main/resources/META-INF/resources/frontend/FlowClient.js",
-    "_cp_to_tgt": "ncp src/main/resources/META-INF/resources/frontend/FlowClient.js target/classes/META-INF/resources/frontend/FlowClient.js",
-    "client": "npm run _cp_to_src && npm run _wrap_fnc && npm run _cp_to_tgt",
+    "client": "node scripts/client.js",
     "webpack": "webpack --config=webpack.tests.config.js",
     "build": "npm run client && tsc",
     "compile": "npm run build && npm run lint",

--- a/flow-client/package.json
+++ b/flow-client/package.json
@@ -9,7 +9,7 @@
     "url": "https://github.com/vaadin/flow/issues"
   },
   "scripts": {
-    "lint": "eslint 'src/main/resources/META-INF/resources/frontend/*.js' && tslint 'src/main/resources/META-INF/resources/frontend/*.ts'",
+    "lint": "eslint \"src/main/resources/META-INF/resources/frontend/*.js\" && tslint --project .",
     "client": "node scripts/client.js",
     "webpack": "webpack --config=webpack.tests.config.js",
     "build": "npm run client && tsc",

--- a/flow-client/scripts/client.js
+++ b/flow-client/scripts/client.js
@@ -1,0 +1,41 @@
+/*
+ * Copyright 2000-2020 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+const fs = require('fs');
+
+const fromDir = "target/classes/META-INF/resources/VAADIN/static/client/";
+const fromFileRegex = /^client-.*\.cache\.js$/;
+
+const toSourceDir = "src/main/resources/META-INF/resources/frontend/";
+const toTargetDir = "target/classes/META-INF/resources/frontend/";
+const toFile  = "FlowClient.js";
+
+const fromFileName = fs.readdirSync(fromDir)
+    .find(filename => fromFileRegex.exec(filename));
+
+// Read from target
+let clientSource = fs.readFileSync(fromDir + fromFileName, 'utf8');
+
+// Wrap with ES module export
+clientSource = `export function init() {
+${clientSource}
+};`;
+
+// Write to source
+fs.writeFileSync(toSourceDir + toFile, clientSource, 'utf8');
+
+// Write to target
+fs.writeFileSync(toTargetDir + toFile, clientSource, 'utf8');

--- a/flow-client/scripts/client.js
+++ b/flow-client/scripts/client.js
@@ -34,6 +34,9 @@ clientSource = `export function init() {
 ${clientSource}
 };`;
 
+// Convert CRLF to LF, needed for building on Windows
+clientSource = clientSource.replace(/\r\n/g, '\n');
+
 // Write to source
 fs.writeFileSync(toSourceDir + toFile, clientSource, 'utf8');
 

